mcp/sensors/csbf_dgps.c

```diff
@@ -406,4 +518,181 @@ void * DGPSMonitor(void * arg)
+
+    while (1) {
+        if (first_time) {
+            first_time = 0;
+            if ((returnval = getaddrinfo(NULL, socket_target->port, &hints, &servinfo)) != 0) {
+                blast_info("getaddrinfo: %s\n", gai_strerror(returnval));
+                return NULL;
+            }
+            // loop through all the results and bind to the first we can
+            for (servinfoCheck = servinfo; servinfoCheck != NULL; servinfoCheck = servinfoCheck->ai_next) {
+                // since a success needs both of these, but it is inefficient to do fails twice,
+                // continue skips the second if on a fail
+                if ((sockfd = socket(servinfoCheck->ai_family, servinfoCheck->ai_socktype,
+                        servinfoCheck->ai_protocol)) == -1) {
+                    blast_err("Failed to make socket");
+                    continue;
+                }
+                if (bind(sockfd, servinfoCheck->ai_addr, servinfoCheck->ai_addrlen) == -1) {
+                    close(sockfd);
+                    blast_err("Failed to bind to socket");
+                    continue;
+                }
+                break;
+            }
+            if (servinfoCheck == NULL) {
+                blast_err("Failed to find any sockets");
+                return NULL;
+            }
+            // set the read timeout (if there isn't a message)
+            struct timeval read_timeout;
+            int read_timeout_usec = 1000;
+            read_timeout.tv_sec = 0;
+            read_timeout.tv_usec = read_timeout_usec;
+            setsockopt(sockfd, SOL_SOCKET, SO_REUSEADDR | SO_RCVTIMEO, &read_timeout, sizeof(read_timeout));
+            // now we set up the print statement vars
+            // need to cast the socket address to an INET still address
+            struct sockaddr_in *ipv = (struct sockaddr_in *)servinfo->ai_addr;
+            // then pass the pointer to translation and put it in a string
+            inet_ntop(AF_INET, &(ipv->sin_addr), ipAddr, INET_ADDRSTRLEN);
+        }
+        numbytes = recvfrom(
+            sockfd,
+            nmea_buffer,
+            sizeof(nmea_buffer) + 1,
+            0,
+            (struct sockaddr *) &sender_addr,
+            &addr_len);
+        // we get an error everytime it times out, but EAGAIN is ok, other ones are bad.
+        if (numbytes == -1) {
+            err = errno;
+            if (err != EAGAIN) {
+                blast_info("Errno is %d\n", err);
+                blast_info("Error is %s\n", strerror(err));
+                berror(err, "Recvfrom");
+            }
+        } else {
+            // Tokenize str from UDP packet: the GPS receiver we talk to
+            // stuffs all NMEA sentences into a single packet, so we separate
+            // them here
+            // blast_info("Received UDP packet from GNSS module\n");
+            char* pTok = NULL;
+            char* pSave = NULL;
+            pTok = strtok_r(nmea_buffer, "\r\n", &pSave);
```

Comment:
any specific reason to use the thread-safe version of this? seems fine but I don't get the use in this single-thread variant other than "sure why not"

Reply:
Gotcha, a case of the sprintf snprintf